src/main/java/com/aws/greengrass/clientdevices/auth/DeviceAuthClient.java

```diff
@@ -141,7 +144,8 @@ public boolean canDevicePerform(AuthorizationRequest request) throws Authorizati
             return true;
         }
 
-        return PermissionEvaluationUtils.isAuthorized(request.getOperation(), request.getResource(),
-                groupManager.getApplicablePolicyPermissions(session));
+        return permissionEvaluationUtils.isAuthorized(request.getOperation(), request.getResource(),
+                permissionEvaluationUtils.transformGroupPermissionsWithVariableValue(session,
```

Comment:
while we're at it, we can clean up by making groupManager a dependency of PermissionEvaluationUtils, then we can do something like

```
permissionEvaluationUtils.isAuthorized(request, session)
```

Reply:
lets also take performance into account, e.g. when evaluating w/ same permissions we shouldn't do variable substitution again

Reply:
done

Reply:
> lets also take performance into account, e.g. when evaluating w/ same permissions we shouldn't do variable substitution again

Thinking through this more, variables are substituted based on current session, so a caching solution for this would be a bit more involved, assuming it would even be useful, so lets table for now